api/v1alpha1/addon_types.go

```diff
@@ -191,24 +192,34 @@ func (StorageClassConfig) VariableSchema() clusterv1.VariableSchema {
 	}
 	return clusterv1.VariableSchema{
 		OpenAPIV3Schema: clusterv1.JSONSchemaProps{
-			Type: "object",
+			Type:     "object",
+			Required: []string{"name", "parameters", "reclaimPolicy", "volumeBindingMode"},
 			Properties: map[string]clusterv1.JSONSchemaProps{
 				"name": {
 					Type:        "string",
 					Description: "Name of storage class config.",
+					MinLength:   ptr.To(int64(1)),
 				},
 				"parameters": {
-					Type:                   "object",
-					Description:            "Parameters passed into the storage class object.",
-					XPreserveUnknownFields: true,
+					Type:        "object",
+					Description: "Parameters passed into the storage class object.",
+					AdditionalProperties: &clusterv1.JSONSchemaProps{
+						Type: "string",
+					},
+					Default: variables.MustMarshal(map[string]string{
+						"csi.storage.k8s.io/fstype": "ext4",
+						"type":                      "gp3",
+					}),
```

Comment:
These are not the default parameters for all CSI drivers. This is specific to only AWS

Reply:
OK in that case then we can't do defaulting here and would have to do it in code.